packages/vm/src/vm.ts

```diff
@@ -99,6 +107,14 @@ export class VM {
 
     this._opts = opts
 
+    const keyPair = nacl.box.keyPair()
+    this.publicKey = keyPair.publicKey
+    this.secretKey = keyPair.secretKey
+
+    if (opts.confidential) {
+      this.confidential = true
+    }
```

Comment:
```suggestion
    this.confidential = opts.confidential ?? false
```

Reply:
But I don't think that you really need this since if this version of ethervm gets used, it's because the dev has already requested sapphire.